pandas/tests/io/excel/test_openpyxl.py

```diff
@@ -139,6 +139,7 @@ def test_write_append_mode(ext, mode, expected):
     [
         ("new", 2, ["apple", "banana"]),
         ("replace", 1, ["pear"]),
+        ("overlay", 1, ["pear", "banana"]),
```

Comment:
Can you please add a dedicated test for `overlay`? Maybe something like: first write a dataframe with 100 rows, then write a dataframe with 50 rows but set its start row to 25. Then you can verify whether row 0:25 + 75:100 are from the first dataframe and 25:75 are from the second dataframe.

Reply:
But that is also the case in this test: `"pear"` is from `df1` and `"banana"` is from `df2`

Reply:
Sorry - yes, you are right!

Maybe add a test to verify that `"overlay"` plays nicely with `startrow` and `startcol` of `to_excel`.

Reply:
added